addons/binding/org.openhab.binding.zoneminder/src/main/java/org/openhab/binding/zoneminder/handler/ZoneMinderServerBridgeHandler.java

```diff
@@ -618,276 +606,692 @@ private void onDisconnected() {
 
             if (thingHandler != null) {
                 thingHandler.onBridgeDisconnected(this);
-                logger.debug("{}: onDisconnected(): Bridge - {}, Thing - {}, Thing Handler - {}", getLogIdentifier(),
-                        thing.getBridgeUID(), thing.getUID(), thingHandler);
+                logger.debug("{}: context='onDisconnected': Bridge - {}, Thing - {}, Thing Handler - {}",
+                        getLogIdentifier(), thing.getBridgeUID(), thing.getUID(), thingHandler);
             }
         }
     }
 
-    @Override
-    public void updateAvaliabilityStatus(IZoneMinderConnectionInfo connection) {
+    private int initRetriesCount = 0;
+    private static int initMaxRevoverableRetries = 5;
+    private static int initMaxUnrecoverableRetries = 5;
 
+    private ZoneMinderConnectionStatus verifyBindingConfiguration(ThingStatus currentStatus) {
+        String context = "verifyBindingConfiguration";
         ThingStatus newStatus = ThingStatus.OFFLINE;
         ThingStatusDetail statusDetail = ThingStatusDetail.NONE;
         String statusDescription = "";
 
-        boolean _isOnline = false;
+        ZoneMinderConnectionStatus status = ZoneMinderConnectionStatus.BINDING_CONFIG_INVALID;
+        ZoneMinderBridgeServerConfig config = null;
+        try {
+            // Is it a retry loop? (Is this step already verified?)
+            // Or is there an unrecoverable error?
+            if ((getLastSucceededStatus().greatherThanEqual(ZoneMinderConnectionStatus.BINDING_CONFIG_LOAD_PASSED))
+                    || (getLastSucceededStatus().hasUnrecoverableError())) {
+                return getLastSucceededStatus();
+            }
 
-        ThingStatus prevStatus = getThing().getStatus();
+            // get Bridge Config
+            config = getBridgeConfig();
+
+            // Check if server Bridge configuration is valid
+            if (config == null) {
+                newStatus = ThingStatus.OFFLINE;
+                statusDetail = ThingStatusDetail.CONFIGURATION_ERROR;
+                statusDescription = "Configuration not found";
+                updateStatus(newStatus, statusDetail, statusDescription);
+
+                logger.error("{}: context='{}' state='{}' check='FAILED' - {}", getLogIdentifier(), context,
+                        newStatus.toString(), statusDescription);
```

Comment:
You don't have to call `toString` it will be called automatically, but only when the logging-level actually applies.